Add ExceptionHandler

marl0vv opened the pull request:
Добавил ExceptionHandler, который обрабатывает все исключения
Добавил ConversionTaskNotFoundException, который выбрасывается, когда задан запрос с id, которого нет в базе
Добавил ErrorDto, который принимает сообщение об ошибке из исключения и выводится, как результат запроса

marl0vv committed:
Add ExceptionHandler

Add ExceptionHandler
Add ConversionTaskNotFoundException
Add ErrorDto

RavenVSS commented on src/main/java/com/glamik/webpconverter/exception/ConversionTaskNotFoundException.java:
```diff
@@ -0,0 +1,7 @@
+package com.glamik.webpconverter.exception;
+
+public class ConversionTaskNotFoundException extends Exception {
```
По моему есть дефолтное исключение `NotFoundException`, можно его использовать

RavenVSS commented on src/main/java/com/glamik/webpconverter/exception/ExceptionApiHandler.java:
```diff
@@ -0,0 +1,19 @@
+@RestControllerAdvice
+public class ExceptionApiHandler {
+    @ExceptionHandler(ConversionTaskNotFoundException.class)
+    public ResponseEntity<ErrorDto> notFoundException(ConversionTaskNotFoundException ex) {
```
Попробуй без ResponseEntity, а код прокидывай через `@ResponseStatus`

RavenVSS commented on src/main/java/com/glamik/webpconverter/exception/dto/ErrorDto.java:
```diff
@@ -0,0 +1,12 @@
+
+@Getter
+@Setter
+@AllArgsConstructor
```
nit: У этой аннотации есть параметр `staticName`, почитай про него, попробуй без `new` создавать это Dto

RavenVSS commented on src/main/java/com/glamik/webpconverter/service/ConversionTaskService.java:
```diff
@@ -55,8 +56,12 @@ public void setConversionErrorStatus(@NonNull UUID id, @NonNull ConversionTaskSt
-    public ConversionTask getConversionTask(@NonNull UUID id) {
-        return conversionTaskRepository.getById(id);
+    public ConversionTask getConversionTask(@NonNull UUID id) throws ConversionTaskNotFoundException {
+        ConversionTask conversionTask = conversionTaskRepository.getById(id);
```
Используй встроенный метод в репозиторий `findById`, и через объект `Optional` вызывай исключение  

nit: Также можно `trows` не указывать, оно без него обрабатывается нормально

marl0vv committed:
Change notFoundException to use NoSuchElementException

marl0vv committed:
Add DirectoryCreationException

Add exception that can be catched in a File Service when creating directories

marl0vv committed:
Add exception when getting an image which conversion went wrong

marl0vv committed:
Add logging to file deletion

RavenVSS commented on src/main/java/com/glamik/webpconverter/exception/ExceptionApiHandler.java:
```diff
@@ -0,0 +1,39 @@
+    @ResponseStatus(code = HttpStatus.INTERNAL_SERVER_ERROR)
+    public ErrorDto conversionStatusIsErrorException(ConversionStatusIsErrorException e) {
+        return ErrorDto.of(e.getMessage());
+    }
```
Надо ещё сделать, чтобы в случае других ошибок кидалось ErrorDto, например можно обрабатывать `Exception.class`

marl0vv committed:
Add default exception handling

marl0vv committed:
Add logger to exceptions

marl0vv committed:
Fix minor things

marl0vv committed:
Change getById to findById

Ok. At some point at the start of the project I declared Repository with id as a Long type. This was a mistake because I use UUID and now I need to change in everywhere.

marl0vv committed:
Delete some unused code
